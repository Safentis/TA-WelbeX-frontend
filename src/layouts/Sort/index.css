.sort {
    align-items: center;
    display: flex;
}

.sort__item {
    align-items: center;
    display: grid;
    justify-content: center;
    transition: .3s;
}

.sort__submenu {
    width: 100%;
    grid-template-columns: 1fr 1fr 1fr;
}

.sort__item_noactive {
    opacity: 0;
    transform: translateX(-10rem);
    z-index: -10;
}

.sort__item_active {
    background-color: var(--main-grey-color);
    opacity: 1;
    transform: translateX(0rem);
    z-index: 0;
}

.sort__nested {
    display: flex;
    justify-self: center;
}