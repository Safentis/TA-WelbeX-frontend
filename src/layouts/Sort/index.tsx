import React, { FC, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFilter } from '@fortawesome/free-solid-svg-icons';

import Button from '../../components/Button/Button';

import './index.css';
import { Props } from './index.interface';
import SortColumn from './SortColumn/SortColumn';
import SortSignal from './SortSignal/SortSignal';
import SortSearch from './SortSearch/SortSearch';

/**
 * Sort components that contains all options of
 * sort, you may choose filter by 3 stage
 * 1) column
 * 2) signals (more, lest, equal and keep)
 * 3) text field
 * @returns {React.ReactElement} - returns react element
 */

const Sort: FC<Props> = ({}): React.ReactElement => {
  //* ----------------------------------------------
  //* Show or hide sub menu with filter
  const [isOpen, setIsOpen] = useState(false);
  const handleClick = (): void => {
    setIsOpen(!isOpen);
  };

  //* Dynamic styles
  const sortItemActiveCls: string = isOpen ? 'sort__item_active' : 'sort__item_noactive';
  const btnActiveCls: string = isOpen ? 'button_active' : '';

  return (
    <ul className="sort">
      <li className="sort__item">
        <Button style={btnActiveCls} handleClick={handleClick}>
          <FontAwesomeIcon icon={faFilter} />
        </Button>
      </li>
      <li className={'sort__item sort__submenu ' + sortItemActiveCls}>
        <SortColumn />
        <SortSignal />
        <SortSearch />
      </li>
    </ul>
  );
};

export default Sort;
