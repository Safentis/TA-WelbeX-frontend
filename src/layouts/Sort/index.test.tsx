import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Sort from './index';

import { Props } from './index.interface';

const setUp = (props: Props) => {
  return shallow(<Sort {...props} />);
};

describe('<Sort />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;

  beforeEach(() => {
    component = setUp({});
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 ul element', () => {
      expect(component.find('ul')).toBeTruthy();
    });

    it('has 1 li element', () => {
      expect(component.find('li')).toBeTruthy();
    });

    it('has 1 Button element', () => {
      expect(component.find('Button')).toBeTruthy();
    });

    it('has 1 FontAwesomeIcon element', () => {
      expect(component.find('FontAwesomeIcon')).toBeTruthy();
    });
  });
});
