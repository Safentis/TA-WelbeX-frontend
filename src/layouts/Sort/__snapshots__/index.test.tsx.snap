// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sort /> component does match to the snapshot 1`] = `
<ul
  className="sort"
>
  <li
    className="sort__item"
  >
    <Button
      handleClick={[Function]}
      style=""
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f0b0",
              "M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z",
            ],
            "iconName": "filter",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Button>
  </li>
  <li
    className="sort__item sort__submenu sort__item_noactive"
  >
    <SortColumn />
    <SortSignal />
    <SortSearch />
  </li>
</ul>
`;
